### ISSUE TITLE
Sudo permissions not assigned on quickstart deployment to non-aws images

### ISSUE THREAD

**Issue by rajinator** [hortonworks/cbd-quickstart#6]
Hi,
I was going through the azure quickstart template for GA release 2.7.1, because some of my deployments were failing. Upon looking at https://github.com/hortonworks/cbd-quickstart/blob/2.7.1/azure/mainTemplate.json

I see that instead of using https://github.com/hortonworks/cbd-quickstart/blob/2.7.1/azure/install-cbd.sh in the azure subfolder, the install script extension is using https://github.com/hortonworks/cbd-quickstart/blob/2.7.1/install-and-start-cbd.sh from the main directory.

In the file, I see that the main() if loop does not have a function similar to aws_prepare_os_user() for assigning sudo permissions to non-aws deployments.

```
main() {
    init

    if [[ -d "${AWS_BIN_LOCATION}" ]]; then
        get_aws_metadata_profile
        aws_prepare_os_user
        : ${WAIT_HANDLE_URL:? required}
    else
        custom_data
    fi

    download_cbd
    download_cb_cli
    set_perm
    su $OS_USER -c "install_cbd"
}
```

```
aws_prepare_os_user() {
    mkdir -p /home/$OS_USER/.ssh
    cp /home/ec2-user/.ssh/authorized_keys /home/$OS_USER/.ssh/
    chown -R $OS_USER:$OS_USER /home/$OS_USER/.ssh
    service sshd reload
    echo "# This is generated by the Cloudbreak deployment's cloud-init script" >> /etc/sudoers.d/cloudbreak
    echo "cloudbreak ALL=(ALL) NOPASSWD:ALL" >> /etc/sudoers.d/cloudbreak
}
```

 I have created a PR from my fork as a potential fix with a non_aws_prepare_os_user function.

```
non_aws_prepare_os_user() {
    echo "# This is generated by the Cloudbreak deployment's cloud-init script" >> /etc/sudoers.d/cloudbreak
    echo "cloudbreak ALL=(ALL) NOPASSWD:ALL" >> /etc/sudoers.d/cloudbreak
}
```

Will link the PR here, please see if this is a valid fix

**Comment by biharitomi** [comment]
Hi @rajinator 

I checked and the Azure template is working for me properly. The deployment was created cbd, cb cli tool installed, and the deployment is running without any errors.

```
[cloudbreak@cbdeployervm cloudbreak-deployment]$ cbd ps
        Name                      Command               State                                                                   Ports
----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
cbreak_cloudbreak_1    /start_cloudbreak_app.sh bash    Up       0.0.0.0:8080->8080/tcp
cbreak_commondb_1      /bin/bash -c cd /var/lib/p ...   Up       172.17.0.1:5432->5432/tcp
cbreak_consul_1        /bin/consul agent -server  ...   Up       8300/tcp, 8301/tcp, 8301/udp, 8302/tcp, 8302/udp, 0.0.0.0:8400->8400/tcp, 0.0.0.0:8500->8500/tcp, 8600/tcp, 172.17.0.1:53->8600/udp
cbreak_haveged_1       haveged -F                       Up
cbreak_identity_1      /tmp/run.sh                      Up       0.0.0.0:8089->8080/tcp
cbreak_logrotate_1     /start.sh                        Up
cbreak_logsink_1       socat -u TCP-LISTEN:3333,r ...   Up       0.0.0.0:32768->3333/tcp
cbreak_logspout_1      /bin/sh -c sleep 1; (ROUTE ...   Up       0.0.0.0:8000->80/tcp
cbreak_mail_1          /bin/sh -c /opt/install.sh ...   Up       172.17.0.1:25->25/tcp
cbreak_periscope_1     /start_periscope_app.sh          Up       0.0.0.0:8085->8080/tcp
cbreak_registrator_1   /bin/registrator consul:// ...   Up
cbreak_smartsense_1    /var/lib/smartsense/start- ...   Exit 1
cbreak_sultans_1       /hortonworks-cloud-auth/st ...   Up       0.0.0.0:3001->3000/tcp, 3001/tcp
cbreak_traefik_1       /entrypoint.sh --debug --w ...   Up       0.0.0.0:443->443/tcp, 0.0.0.0:80->80/tcp, 172.17.0.1:8081->8080/tcp
cbreak_uluwatu_1       /hortonworks-cloud-web/sta ...   Up       0.0.0.0:3000->3000/tcp
[cloudbreak@cbdeployervm cloudbreak-deployment]$
[cloudbreak@cbdeployervm cloudbreak-deployment]$
[cloudbreak@cbdeployervm cloudbreak-deployment]$
[cloudbreak@cbdeployervm cloudbreak-deployment]$ cb --version
cb version 2.7.1-2018-07-12T09:18:16
```

That you mentioned [install-cbd.sh file](https://github.com/hortonworks/cbd-quickstart/blob/2.7.1/azure/install-cbd.sh) has been left on the repo mistakenly due to the marketplace approach that still needs to be implemented in this repo for Azure. On the other hand that script also not adds the "cloudbreak" user to the sudoers and on the created VM the "cloudbreak" user is fully functional as it is able to manage the deployment and also has root permissions.

AWS is a different story, because the VM is created with "ec2-user" by default and we needed a workaround to have  the same "cloudbreak" user on all of the providers' quickstart option.

Br,
Tamas

**Comment by rajinator** [comment]
Hi @biharitomi ,

I also changed the admin username in the template parameters form on Azure from cloudbreak to something else. Could this be a potential problem?

Regards,
Raj

**Comment by biharitomi** [comment]
Hi @rajinator ,

Yes, it could be an issue.
All of the providers use "cloudbreak" user by default, only on Azure you could change it through the templating mechanism.

Why should we add the user to the sudoers when it has root permissions on the Azure VM by default?

Br,
Tamas

**Comment by biharitomi** [comment]
Hi @rajinator ,

We found an issue if you change the username in the templates then our script doesn't handle the docker permissions correctly.
We will fix this soon from the 2.4 line.

Br,
Tamas

**Comment by pdarvasi** [comment]
It is fixed and backported to 2.4.3 and 2.7.1
https://github.com/hortonworks/cbd-quickstart/commit/37f24e6e8ec23e69781397a7f9ddfc0a56a6a08d

**Comment by rajinator** [comment]
Awesome, thanks @biharitomi @pdarvasi !
